lib/aot/compiler.cpp

```diff
@@ -4590,9 +4598,18 @@ Expect<void> outputNativeLibrary(const std::filesystem::path &OutputPath,
       },
 #elif WASMEDGE_OS_LINUX
   lld::elf::link(
-      std::initializer_list<const char *>{"ld.lld", "--shared", "--gc-sections",
-                                          "--discard-all", ObjectName.c_str(),
-                                          "-o", OutputPath.u8string().c_str()},
+      std::initializer_list<const char *> {
+        "ld.lld", "--shared", "--gc-sections",
+#if defined(__x86_64__)
+            "-L/usr/lib/gcc/x86_64-linux-gnu/9", "-latomic",
+#elif defined(__aarch64__)
+            "-L/usr/lib/gcc/aarch64-linux-gnu/9", "-latomic",
```

Comment:
This is a temporary hack on linux platform. When compile with atomic instrcutions in LLVM IR, the object file should link against `__atomic_*` [1] in `libatomic.a`. I have tried to link a object file `__atomic_load` with clang by adding `-latomic` to clang, the clang will automatically generate multi-platfrom library path (I guess its implementation is [2]) and find libatomic. I'm not sure how to generate libatomic library path in our AOT compiler.

[1] https://llvm.org/docs/Atomics.html#libcalls-atomic
[2] https://github.com/llvm-mirror/clang/blob/master/lib/Driver/ToolChains/Linux.cpp

Reply:
You have to set proper alignment restriction to avoid llvm generating calls to libatomic, in my own experience.